Challo/Challo/Modules/Common/Bookings/BookingAPIParser.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  BookingAPIParser.swift
+//  Challo
+//
+//  Created by Tan Le Yang on 30/3/21.
+//
+
+import Foundation
+
+class BookingAPIParser: APIParser {
+
+    typealias JSON = NetworkManager.JSON
+
+    func extractBookingsJSON(response: JSON) -> [JSON] {
+        guard let data = response["data"],
+              let bookingsInfo = data as? [JSON] else {
+            return []
+        }
+        
+        return bookingsInfo
+    }
+
+    func extractTrailID(json: JSON) -> UUID? {
+        UUID(uuidString: json[Key.trailId] as? String ?? "")
+    }
+
+    func extractTouristID(json: JSON) -> UUID? {
+        UUID(uuidString: json[Key.touristId] as? String ?? "")
+    }
+
+    func extractGuideID(json: JSON) -> UUID? {
+        UUID(uuidString: json[Key.guideId] as? String ?? "")
+    }
+    
+    func convertJSONToBooking(json: JSON, guide: Guide, trail: Trail, tourist: Tourist) -> Booking? {
+        guard let bookingID = UUID(uuidString: json[Key.bookingId] as? String ?? ""),
+              let createdAtString = json[Key.createdAt] as? String,
+              let dateString = json[Key.date] as? String,
+              let statusString = json[Key.status] as? String else {
+            return nil
+        }
+    
+        guard let date = Date.construct(with: dateString),
+              let createdAt = Date.construct(with: createdAtString),
+              let bookingStatus = BookingStatus(rawValue: statusString) else {
+            return nil
+        }
+    
+        let fee = convertJSONDoubleValueToDouble(json[Key.fee])
+    
+        return Booking(bookingId: bookingID,
+                       fee: fee,
+                       date: date,
+                       createdAt: createdAt,
+                       status: bookingStatus,
+                       guide: guide,
+                       trail: trail,
+                       tourist: tourist,
+                       review: nil)
```

Comment:
no review ah? Or not available in api yet

Reply:
Btw, reviews for bookings are found in GET /review routes. Would you prefer review to be included in GET booking?

Reply:
Oh I was supposed to add it in, missed it out ps. @shaoyi1997 nah it's fine!